src/ProducerEventProcessor/ProducerEventProcessor.cpp

```diff
@@ -169,8 +172,56 @@ class ProducerEventProcessorImpl : public ProducerEventProcessor {
   // <producer_id, producer_string_id> -> client_string_id
   absl::flat_hash_map<std::pair<uint64_t, uint64_t>, uint64_t>
       producer_interned_string_id_to_client_string_id_;
+
+  // Needed to allow merging between call stacks and tracepoints, see design doc:
```

Comment:
This sentence is partly misleading as here we are merging the thread state slices with the callstacks.